### PR TITLE
Allow experiment runner access to non public entities

### PR DESCRIPTION
**Pull request by helenst**
Addresses #55 

Provide special access on the entity archive view so that the experiment runner can download non-public entities. This introduces the `RunningExperiment` model which keeps track of information about experiments that have not yet completed. The model provides a unique UUID which acts as the signature sent to the experiment runner, and must be passed in an `Authorization` header when attempting to access the entity download.

### CODE REVIEW

File: weblab/experiments/tests/test_processing.py [hunk 1]
```diff
@@ -227,6 +242,26 @@ def test_records_finished_status(self, returned_status, stored_status,
         assert queued_experiment.status == stored_status
         assert queued_experiment.return_text == 'finished'
 
+    @pytest.mark.parametrize('returned_status', [
+        'success',
+        'partial',
+        'failed',
```
[the remark this answers is not in the document]
Comment (jonc125) [reply weblab/experiments/tests/test_processing.py]: I've implemented this.

File: weblab/experiments/processing.py [hunk 2]
```diff
@@ -54,8 +53,10 @@ def submit_experiment(model, model_version, protocol, protocol_version, user):
         experiment=experiment,
         author=user,
     )
+    version.save()
```
Comment (helenst) [inline weblab/experiments/processing.py]: `objects.create` will write to the database so `save` should not be necessary

File: weblab/experiments/processing.py [hunk 3]
```diff
@@ -54,8 +53,10 @@ def submit_experiment(model, model_version, protocol, protocol_version, user):
...
 
     run = RunningExperiment.objects.create(experiment_version=version)
+    run.save()
```
Comment (helenst) [inline weblab/experiments/processing.py]: ... and the same here

File: weblab/experiments/models.py [hunk 4]
```diff
@@ -129,7 +129,8 @@ def archive_path(self):
 
     @property
     def signature(self):
-        return str(self.id)
+        running = self.running.first()
+        return str(running.id) if running else ''
```
[the remark this answers is not in the document]
Comment (helenst) [reply weblab/experiments/models.py]: That's probably better... I'll add this.

**Pull request merged**